From: OpenWrt Builder <openwrt@openwrt.org>
Date: Wed, 2 Oct 2025 19:25:00 +0200
Subject: [PATCH] arm64: dts: qcom: msm8916: add reboot-mode support

Add syscon-reboot-mode node to enable software reboot to fastboot
and EDL mode on MSM8916 devices. Also update tcsr compatible to use
generic syscon driver with simple-mfd.

Signed-off-by: OpenWrt Builder <openwrt@openwrt.org>
---
 arch/arm64/boot/dts/qcom/msm8916.dtsi | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

--- a/arch/arm64/boot/dts/qcom/msm8916.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916.dtsi
@@ -1519,8 +1519,8 @@
 		};
 
 		tcsr: syscon@1937000 {
-			compatible = "qcom,tcsr-msm8916", "syscon";
-			reg = <0x01937000 0x30000>;
+			compatible = "syscon", "simple-mfd";
+			reg = <0x1937000 0x30000>;
 		};
 
 		mdss: display-subsystem@1a00000 {
@@ -2228,6 +2228,17 @@
 		};
 	};
 
+	reboot-mode {
+		compatible = "syscon-reboot-mode";
+		syscon = <&tcsr>;
+		offset = <0x6100>;
+
+		mode-normal = <0x00000000>;
+		mode-bootloader = <0x77665500>;
+		mode-recovery = <0x77665502>;
+		mode-edl = <0x00000001>;
+	};
+
 	timer {
 		compatible = "arm,armv8-timer";
 		interrupts = <GIC_PPI 2 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_LOW)>,